core/src/main/java/software/amazon/awssdk/protocol/json/internal/SimpleTypeJsonMarshallers.java

```diff
@@ -125,7 +123,7 @@ public void marshall(List list, StructuredJsonGenerator jsonGenerator, JsonMarsh
 
         @Override
         protected boolean shouldEmit(List list) {
-            return !(list.isEmpty() && list instanceof SdkInternalList && ((SdkInternalList) list).isAutoConstruct());
+            return list != null;
```

Comment:
I believe we can just omit shouldEmit, null fields aren't serialized.  The old code was relying on this since we don't do a null check here.

https://github.com/aws/aws-sdk-java/blob/8976a3596d1be66f688256ce4907b8d286202e8a/aws-java-sdk-core/src/main/java/com/amazonaws/protocol/json/internal/SimpleTypeJsonMarshallers.java#L33